### PR TITLE
build: create sourcemap files

### LINKED ISSUE #852: Sourcemap paths are incorrect
#### Bug, feature request, or proposal:

Bug
#### What is the current behavior?

@jelbourn in publishing the latest version to npm, it seems you left your own project path in the javascript map files.  Consider the following excerpt downloaded directly from npm:

node_modules/@angular2-material/button/button.spec.js.map:

```
{"version":3,"file":"button.spec.js","sourceRoot":"","sources":["../../Users/jelbourn/material2/tmp/broccoli_type_script_compiler-input_base_path-ln
```

### PR DESCRIPTION
**Pull request by DevVersion**
* Creates sourceMap files for all bundles inside of the release.

<details>
screen shot 2017-04-10 at 11 40 02

</details>

<p>

**Note**: Tested all maps using `source-map-explorer`. Also uses tools as in [angular/angular#build.sh](https://github.com/angular/angular/blob/master/build.sh)

</p>
Fixes #852

### CODE REVIEW

File: tools/gulp/tasks/library.ts [hunk 1]
```diff
@@ -92,3 +101,23 @@ task('library:assets', ['library:assets:scss', 'library:assets:html']);
 task('library:assets:scss', sassBuildTask(materialDir, libraryRoot, true));
 task('library:assets:html', copyTask(join(libraryRoot, '**/*.+(html|scss)'), materialDir));
 task('library:assets:inline', () => inlineResources(materialDir));
+
+/**
+ * Finds the original sourcemap of the file and maps it to the current file.
+ * This is useful when multiple transformation happen (e.g TSC -> Rollup -> Uglify)
+ **/
+async function remapSourcemap(sourceFile: string) {
+  (await sorcery.load(sourceFile)).write();
```
Comment (jelbourn) [inline tools/gulp/tasks/library.ts]: Any idea how this actually works?

Comment (DevVersion) [inline tools/gulp/tasks/library.ts]: Yeah it recursively goes through the parent's sourcemap file. As from the package, it follows the chain of source-files back to the first one. But not worth mentioning how the package works, I think the JSDoc explains what it does.

**Comment by jelbourn** [comment]
@DevVersion can you rebase?

**Comment by DevVersion** [comment]
@jelbourn Done. Pretty cool that since we have our shared `package-build` functions the source-maps are also built for the examples :tada:

**Pull request merged**